src/Sylius/Bundle/CoreBundle/Tests/Form/Type/CatalogPromotion/CatalogPromotionActionTypeTest.php

```diff
@@ -0,0 +1,335 @@
+
+        $this->channelRepository->findAll(Argument::any())->willReturn(
+            [$this->channel->reveal()]
+        );
+
+        $form = $this->factory->create(CatalogPromotionActionType::class, $fixedDiscount);
+        $form->submit([
+            'type' => 'percentage_discount',
+            'configuration' => [
+                'amount' => ''
+            ]
+        ]);
+
+        $this->assertTrue($form->isSynchronized());
+        $catalogPromotionAction = $form->getData();
+
+        $this->assertInstanceOf(CatalogPromotionActionInterface::class, $catalogPromotionAction);
+        $this->assertSame('percentage_discount', $catalogPromotionAction->getType());
+        $this->assertSame(['amount' => null], $catalogPromotionAction->getConfiguration());
+    }
+
+    /** @test */
+    public function it_changes_action_to_fixed_discount_with_empty_amount(): void
+    {
+        $percentageDiscount = $this->setupPercentageDiscount();
+
+        $this->channelRepository->findAll(Argument::any())->willReturn(
+            [$this->channel->reveal()]
+        );
+
+        $form = $this->factory->create(CatalogPromotionActionType::class, $percentageDiscount);
+        $form->submit([
+            'type' => 'fixed_discount',
+            'configuration' => [
+                'WEB_US' => [
+                    'amount' => ''
+                ]
+            ]
+        ]);
+
+        $this->assertTrue($form->isSynchronized());
+        $catalogPromotionAction = $form->getData();
+
+        $this->assertInstanceOf(CatalogPromotionActionInterface::class, $catalogPromotionAction);
+        $this->assertSame('fixed_discount', $catalogPromotionAction->getType());
+        $this->assertSame(
+            [
+                'WEB_US' => [
+                    'amount' => null
+                ]
+            ],
+            $catalogPromotionAction->getConfiguration()
+        );
+    }
+
+    /** @test */
+    public function it_updates_fixed_discount_with_not_valid_amount(): void
+    {
+        $fixedDiscount = $this->setupFixedDiscount();
+
+        $this->channelRepository->findAll(Argument::any())->willReturn(
+            [$this->channel->reveal()]
+        );
+
+        $form = $this->factory->create(CatalogPromotionActionType::class, $fixedDiscount);
+        $form->submit([
+            'type' => 'fixed_discount',
+            'configuration' => [
+                'WEB_US' => [
+                    'amount' => 'Not valid amount'
+                ]
+            ]
+        ]);
+
+        $this->assertTrue($form->isSynchronized());
+        $catalogPromotionAction = $form->getData();
+
+        $this->assertInstanceOf(CatalogPromotionActionInterface::class, $catalogPromotionAction);
+        $this->assertSame('fixed_discount', $catalogPromotionAction->getType());
+        $this->assertSame(
+            [
+                'WEB_US' => [
+                    'amount' => null
+                ]
+            ],
```

Comment:
I would make it `['WEB_US' => ['amount' => null]],` here and in other places, but it's probably a matter of taste 🖖